Cargo.toml

```diff
@@ -41,6 +41,9 @@ clap = { version = "4.4.18", features = ["derive", "env"] }
 anyhow = "1.0.79"
 async-trait = "0.1.77"
 async-channel = "2.1.1"
+async-std = { version = "1.12.0", features = ["attributes"] }
+libp2p = { version = "0.52", features = ["tcp", "tls", "dns", "async-std", "noise", "yamux", "websocket", "ping", "macros"] }
```

Comment:
Are async-std and futures being used?

Reply:
will be on the next step which is the communication itself, I meant to add just the backbone (I can remove until I implement it if needed)